regparser/notice/compiler.py

```diff
@@ -258,27 +258,29 @@ def replace_node_and_subtree(self, node):
         else:
             parent = self.get_parent(node)
 
-        prev_idx = [idx for idx, c in enumerate(parent.children)
-                    if c.label == node.label]
-        if prev_idx:
-            # replace existing element in place
-            prev_idx = prev_idx[0]
-            parent.children = (parent.children[:prev_idx] + [node] +
-                               parent.children[prev_idx + 1:])
-        else:
-            # actually adding a new element
-            parent.children = self.add_child(parent.children, node,
-                                             getattr(parent, 'child_labels',
-                                                     []))
-
-        # Finally, we see if this node is the parent of any 'kept' children.
-        # If so, add them back
-        label_id = node.label_id()
-        if label_id in self._kept__by_parent:
-            for kept in self._kept__by_parent[label_id]:
-                node.children = self.add_child(node.children, kept,
-                                               getattr(node, 'child_labels',
-                                                       []))
+        # not sure what to do if parent is null....
```

Comment:
Tracing the code a bit, if parent is null, it means that the *expected* parent label id couldn't be found in the tree. That most likely means that the location of changes to reg J (in final rules) aren't being parsed correctly.

I'd recommend debugging by printing out what the current label is and what parent label it's looking for. If the current label is "1010--b", you've got bigger problems (as it doesn't know what section you're in).